src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -2595,7 +2595,14 @@ private void doOpenSourceFile(final FileSystemItem file,
          @Override
          public void execute(EditingTarget target)
          {
-            if (position != null)
+            // the rstudioapi package can use the proxy (-1, -1) position to
+            // indicate that source navigation should not occur; ie, we should
+            // preserve whatever position was used in the document earlier
+            boolean navigateToPosition =
+                  position != null &&
+                  (position.getLine() != -1 && position.getColumn() != -1);
```

Comment:
Nit: Doesn't this treat any position with a `-1` component as non-navigable rather than the `-1, -1` special pair?